### PR TITLE
Dirk dahlquist

### LINKED ISSUE #8023: Tempus:  Add Dahlquist test for DIRK AppAction
## Enhancement
@trilinos/tempus 

Add unit testing for DIRK AppAction similar to the Dahlquist test for Explicit RK AppAction.  This can use the EDIRK Trapezoidal stepper to test with and without FSAL, Embedded, and step failure.

### PR DESCRIPTION
**Pull request by sconde**
@trilinos/tempus 

## Related Issues

* Closes #8023

### CODE REVIEW

File: packages/tempus/test/TestModels/DahlquistTestModel_impl.hpp [hunk 1]
```diff
@@ -180,12 +202,30 @@ evalModelImpl(
...
+
+  //TODO: implement for DIRK
+
+  volatile auto x_in_ = inArgs.get_x();
+
+
+
+  //TODO: implement for DIRK
+
+  volatile auto x_in_ = inArgs.get_x();
```
Comment (ccober6) [inline packages/tempus/test/TestModels/DahlquistTestModel_impl.hpp:221]: Seems to be a duplicate definition?

File: packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  double dt = 1.0;
+  solutionHistory->getWorkingState()->setTimeStep(dt);
+  solutionHistory->getWorkingState()->setTime(dt);
+  stepper->takeStep(solutionHistory);           // Primary testing occurs in
+                                                // modifierX during takeStep().
+  // Test stepper properties.
+  TEUCHOS_ASSERT(stepper->getOrder() == 2);
+}
+
+} // namespace Tempus_Test
```
Comment (ccober6) [inline packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp:181]: It looks like this is focused on SDIRK21.  Should it be in Tempus_UnitTest_SDIRK_21Pair.cpp?

Comment (sconde) [reply packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp:185]: I was hoping to add EDIRK Trapezoid to this test as well, per the comment in the issue. What do you think?

Comment (ccober6) [reply packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp:185]: I guess I would put that in Tempus_UnitTest_EDIRK_TrapezoidalRule.cpp.  I was thinking we should group by Stepper.

Comment (sconde) [reply packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp:343]: I've just pushed the ESDIRK_Trapezoid test section

Comment (ccober6) [reply packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp:343]: Do you want to put these in this file or in Tempus_UnitTest_SDIRK_21Pair.cpp and Tempus_UnitTest_EDIRK_TrapezoidalRule.cpp?

Comment (sconde) [reply packages/tempus/unit_test/Tempus_UnitTest_DIRK.cpp]: Done

**Review by ccober6 — APPROVED** [review]
Looks good!

File: packages/tempus/unit_test/Tempus_UnitTest_EDIRK_TrapezoidalRule.cpp [hunk 3]
```diff
@@ -64,4 +65,163 @@ TEUCHOS_UNIT_TEST(EDIRK_TrapezoidalRule, AppAction)
 }
 
 
+// ************************************************************
+// ************************************************************
+
+class StepperRKModifierEDIRK_TrapezoidaTest
```
Comment (ccober6) [inline packages/tempus/unit_test/Tempus_UnitTest_EDIRK_TrapezoidalRule.cpp:71]: There seems to be several places where "Trapezoida" is used.  Should it be "Trapezoidal"?

**Review by ccober6 — APPROVED** [review]
Looks good!

**Pull request merged**